### PR TITLE
Refactors synchronizer.js file to include a class

### PR DESCRIPTION
**Pull request by jaysaleh**
This pull request refactors `synchronizer.js`  to include a synchronizer object

### CODE REVIEW

File: src/main/webapp/synchronizer.js [hunk 1]
```diff
@@ -19,26 +19,28 @@ let lastTime;
 
 const TIME_INTERVAL_MS = 100;
```
Comment (amuamushu) [inline src/main/webapp/synchronizer.js:20]: should this constant be inside the class?

File: src/main/webapp/view/video/video.js [hunk 2]
```diff
@@ -45,7 +43,7 @@ function onYouTubeIframeAPIReady() {
 /** `event` plays the YouTube video. */
 function onPlayerReady(event) {
   event.target.playVideo();
-  startVideoSyncTimer();
+  window.synchronizer.startVideoSyncTimer();
```
Comment (f-camp) [inline src/main/webapp/view/video/video.js:46]: Why window here? Wht not pass this into the video.js class? If this is the plan please add a TODO.

Comment (jaysaleh) [reply src/main/webapp/view/video/video.js:46]: A TODO for this is located in `View.js` where `synchronizer` is created. Do I need to add a TODO everywhere I do `window.synchronizer`?

File: src/main/webapp/synchronizer.js [hunk 3]
```diff
@@ -19,26 +19,28 @@ let lastTime;
...
-/**
- * Starts timer which broadcasts current video time every
- * `TIME_INTERVAL` milliseconds.
- */
-export function startVideoSyncTimer() {
-  window.setInterval(() => {
-    sync(window.videoPlayer.getCurrentTime());
-  }, /* ms= */ TIME_INTERVAL_MS);
-}
+export default class Synchronizer {
```
Comment (billie-cat) [inline src/main/webapp/synchronizer.js:22]: Remember that classes should have JsDocs

https://google.github.io/styleguide/jsguide.html#jsdoc-class-comments

File: src/main/webapp/synchronizer.js [hunk 4]
```diff
@@ -19,26 +19,28 @@ let lastTime;
...
-/**
- * Calls functions that seek transcript, and discussion to `currentTime`
- * (number of seconds since start of video), when the `currentTime`
- * changes from the last time this was called.
- */
-function sync(currentTime) {
-  if (currentTime == lastTime) {
-    return;
+  /**
+   * Calls functions that seek transcript, and discussion to `currentTime`
```
Comment (billie-cat) [inline src/main/webapp/synchronizer.js:34]:
```suggestion
   * Calls functions that seek transcript and discussion to `currentTime`
```

File: src/main/webapp/synchronizer.js [hunk 5]
```diff
@@ -19,26 +19,28 @@ let lastTime;
...
+   * (number of seconds since start of video), when the `currentTime`
```
Comment (billie-cat) [inline src/main/webapp/synchronizer.js:35]: current time is in seconds? Why is that?

File: src/main/webapp/view/view.js [hunk 6]
```diff
@@ -11,6 +11,9 @@
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
 // limitations under the License.
+// module 'view.js'
```
Comment (billie-cat) [inline src/main/webapp/view/view.js:14]: What is this comment for? If it is necessary I would add a new line between this comment and the copyright or else it gets lost with the copyright.

Comment (jaysaleh) [reply src/main/webapp/view/view.js:14]: Sorry this is an error

File: src/main/webapp/synchronizer.js (hunk 5 above)
Comment (jaysaleh) [reply src/main/webapp/synchronizer.js:35]: The video player returns the current time in seconds from the start of the video, the transcript and the discussion areas take the seconds and convert to ms. We will be converting this to ms in the future.

Added a TODO and #267

Comment (jaysaleh) [reply src/main/webapp/synchronizer.js:36]: @n-wach is working on this right now

File: src/main/webapp/view/video/video.js (hunk 2 above)
Comment (f-camp) [reply src/main/webapp/view/video/video.js:46]: Ah I see, as long as there is a TODO.

File: src/main/webapp/synchronizer.js (hunk 5 above)
Comment (billie-cat) [reply src/main/webapp/synchronizer.js:36]: Todo sounds great!

File: src/main/webapp/synchronizer.js [hunk 7]
```diff
@@ -19,26 +19,30 @@ let lastTime;
...
-/**
- * Starts timer which broadcasts current video time every
- * `TIME_INTERVAL` milliseconds.
- */
-export function startVideoSyncTimer() {
-  window.setInterval(() => {
-    sync(window.videoPlayer.getCurrentTime());
-  }, /* ms= */ TIME_INTERVAL_MS);
-}
+/** Handles when to seek transcript and discussion areas. */
```
Comment (billie-cat) [inline src/main/webapp/synchronizer.js:22]:
```suggestion
/** Handles when to seek transcript and discussion areas according to video time. */
```

**Pull request merged**